teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -0,0 +1,149 @@
+import { announceToScreenReader } from "../transforms/announceToScreenReader";
+import { FocusTrap } from "react-common/components/controls/FocusTrap";
+import css from "./styling/CatalogOverlay.module.scss";
+
+interface CatalogHeaderProps {
+    onClose: () => void;
+}
+const CatalogHeader: React.FC<CatalogHeaderProps> = ({ onClose }) => {
+    return (
+        <div className={css["header"]}>
+            <span className={css["title"]}>{Strings.SelectCriteriaDescription}</span>
+            <Button
+                className={css["close-button"]}
+                rightIcon="fas fa-times-circle"
+                onClick={onClose}
+                title={Strings.Close}
+            />
+        </div>
+    );
+};
+
+interface CatalogItemLabelProps {
+    catalogCriteria: CatalogCriteria;
+    allowsMultiple: boolean;
+    existingInstanceCount: number;
+    recentlyAdded: boolean;
+}
+const CatalogItemLabel: React.FC<CatalogItemLabelProps> = ({
+    catalogCriteria,
+    allowsMultiple,
+    existingInstanceCount,
+    recentlyAdded,
+}) => {
+    const canAddMore = allowsMultiple || existingInstanceCount === 0;
+    const showRecentlyAddedIndicator = recentlyAdded && canAddMore;
+    return (
+        <div className={css["catalog-item-label"]}>
+            <div className={css["action-indicators"]}>
+                {canAddMore ? (
+                    <>
+                        <i
+                            className={classList(
+                                "fas fa-check",
+                                css["recently-added-indicator"],
+                                showRecentlyAddedIndicator ? undefined : css["hide-indicator"]
+                            )}
+                            title={lf("Added!")}
+                        />
+                        <i
+                            className={classList(
+                                "fas fa-plus",
+                                showRecentlyAddedIndicator ? css["hide-indicator"] : undefined
+                            )}
+                            title={Strings.AddToChecklist}
+                        />
+                    </>
+                ) : (
+                    <span className={css["max-label"]}>{Strings.Max}</span>
+                )}
+            </div>
+            <ReadOnlyCriteriaDisplay catalogCriteria={catalogCriteria} showDescription={true} />
+        </div>
+    );
+};
+
+const CatalogList: React.FC = () => {
+    const { state: teacherTool } = useContext(AppStateContext);
+
+    const recentlyAddedWindowMs = 500;
+    const [recentlyAddedIds, setRecentlyAddedIds] = useState<string[]>([]);
+
+    const criteria = useMemo<CatalogCriteria[]>(
+        () => getCatalogCriteria(teacherTool),
+        [teacherTool.catalog, teacherTool.rubric]
+    );
+
+    function onItemClicked(c: CatalogCriteria) {
+        addCriteriaToRubric([c.id]);
+        setRecentlyAddedIds([...recentlyAddedIds, c.id]);
+
+        setTimeout(() => {
```

Comment:
This seems unreliable when adding many in quick succession. May need to adjust approach.

Reply:
Hmm, I see what you're saying. Maybe try a queue? The adding the id can stay the same since it's basically enqueuing the entry in a JavaScript way, but then when removing the id from the list, instead of filtering, just pop the first item off the list. I'm not fully certain how this will look, especially since I've seen `setTimeout` behave buggy before, but it might help.

Reply:
I'm working on this now. Best approach I can think of is to use a map, but the real issue is not so much the data structure. It's the async nature of React state updates, so sometimes one change ends up overwriting another.

Turns out there's a React feature to help with this, where you can pass "[updater functions](https://react.dev/reference/react/useState#passing-the-updater-function)" into set state calls, which accept the previous state as a parameter and modify that instead of acting on the state var directly. This seems to work well. Now I'm just tidying things up before pushing.

Reply:
Oh, nice! My other comments might not be super helpful, then, just a heads-up.